ocaml/xapi-storage-script/main.ml

```diff
@@ -650,8 +648,11 @@ module Attached_SRs = struct
         in
         let dir = Filename.dirname path in
         Sys.mkdir_p dir >>= fun () -> Sys.save path ~contents
-    )
-    >>= fun () -> return ()
```

Comment:
I'm surprised that this can be removed and we still return a monadic value.

Reply:
There are two monads here, the conversion is still done in the `add` function

Reply:
I have updated `update_state_path ()` to make it return a `Lwt_result.t`